homeassistant/components/github/coordinator.py

```diff
@@ -42,80 +45,131 @@ def __init__(
 
 
 class RepositoryInformationDataUpdateCoordinator(
     GitHubBaseDataUpdateCoordinator[GitHubRepositoryModel]
 ):
     """Data update coordinator for repository information."""
 
-    async def fetch_data(self) -> GitHubRepositoryModel:
+    async def fetch_data(self) -> GitHubResponseModel[GitHubRepositoryModel]:
         """Get the latest data from GitHub."""
-        result = await self._client.repos.get(self.repository)
-        return result.data
+        return await self._client.repos.get(self.repository, **{"etag": self._etag})
 
 
 class RepositoryReleaseDataUpdateCoordinator(
     GitHubBaseDataUpdateCoordinator[GitHubReleaseModel]
 ):
     """Data update coordinator for repository release."""
 
-    async def fetch_data(self) -> GitHubReleaseModel | None:
-        """Get the latest data from GitHub."""
-        result = await self._client.repos.releases.list(
-            self.repository, **{"params": {"per_page": 1}}
-        )
-        if not result.data:
+    @staticmethod
+    def _parse_response(
+        response: GitHubResponseModel[GitHubReleaseModel | None],
+    ) -> GitHubReleaseModel | None:
+        """Parse the response from GitHub API."""
+        if not response.data:
             return None
 
-        for release in result.data:
+        for release in response.data:
             if not release.prerelease:
                 return release
 
         # Fall back to the latest release if no non-prerelease release is found
-        return result.data[0]
+        return response.data[0]
+
+    async def fetch_data(self) -> GitHubReleaseModel | None:
+        """Get the latest data from GitHub."""
+        return await self._client.repos.releases.list(
+            self.repository, **{"params": {"per_page": 1}, "etag": self._etag}
+        )
 
 
 class RepositoryIssueDataUpdateCoordinator(
     GitHubBaseDataUpdateCoordinator[IssuesPulls]
 ):
     """Data update coordinator for repository issues."""
 
-    async def fetch_data(self) -> IssuesPulls:
-        """Get the latest data from GitHub."""
-        base_issue_response = await self._client.repos.issues.list(
-            self.repository, **{"params": {"per_page": 1}}
-        )
-        pull_response = await self._client.repos.pulls.list(
-            self.repository, **{"params": {"per_page": 1}}
-        )
+    _issue_etag: str | None = None
+    _pull_etag: str | None = None
 
-        pulls_count = pull_response.last_page_number or 0
-        issues_count = (base_issue_response.last_page_number or 0) - pulls_count
+    @staticmethod
+    def _parse_response(response: IssuesPulls) -> IssuesPulls:
+        """Parse the response from GitHub API."""
+        return response
 
-        issue_last = base_issue_response.data[0] if issues_count != 0 else None
+    async def fetch_data(self) -> IssuesPulls:
+        """Get the latest data from GitHub."""
+        pulls_count, pull_last, issues_count, issue_last = 0, None, 0, None
```

Comment:
This makes it a bit harder to read IMHO.

```suggestion
        pulls_count = 0
        pull_last = None
        issues_count = 0
        issue_last = None
```